processing/src/main/java/io/druid/query/groupby/GroupByQueryEngine.java

```diff
@@ -71,6 +79,62 @@
   private final Supplier<GroupByQueryConfig> config;
   private final StupidPool<ByteBuffer> intermediateResultsBufferPool;
 
+  private static void getBytesFromBuffer(ByteBuffer src, byte[] dst, int srcOffset, int readLen)
+  {
+    for (int i = 0; i < readLen; i++) {
+      dst[i] = src.get(srcOffset + i);
+    }
+  }
+
+  private static final Comparator<ByteBuffer> makeKeyComparator(final List<DimensionHandler> dimHandlers)
+  {
+    Comparator comp = new Comparator<ByteBuffer>()
+    {
+      @Override
+      public int compare(ByteBuffer o1, ByteBuffer o2)
+      {
+        int pos = 0;
+        int limit = o1.limit();
+        int ret = 0;
+        int dimIndex = 0;
+        int maxDimIndex = dimHandlers.size();
+
+        while (pos < limit && dimIndex < maxDimIndex) {
+          DimensionHandler handler = dimHandlers.get(dimIndex);
+          if (handler == null) {
+            // Special time column handling
+            int int1 = o1.getInt(pos);
+            int int2 = o2.getInt(pos);
+            pos += Ints.BYTES;
+            ret = Integer.compare(int1, int2);
+          } else {
+            int valLen = handler.getEncodedValueSize();
+            byte[] bytes1 = new byte[valLen];
+            byte[] bytes2 = new byte[valLen];
+
+            getBytesFromBuffer(o1, bytes1, pos, valLen);
+            getBytesFromBuffer(o2, bytes2, pos, valLen);
+
+            pos += valLen;
+
+            Comparator encodedComparator = handler.getEncodedComparator();
```

Comment:
For this model of doing things, it would likely be better to force the implementations to implement the comparison of two `byte[]` instead of deserializing on comparison.  The code as written is essentially eliminating the ability for a column implementation to optimize itself even if it has the ability to do a byte-wise comparison.